### ISSUE TITLE
Question about Domain Strategy Identity Routing

### ISSUE THREAD

**Issue by asparatu** [Finbuckle/Finbuckle.MultiTenant#680]
Hello Andrew,

I have one more issue I can not understand how to fix. When the AddMultiTenant<TenantInfo>() has WithBasePathStrategy(options => options.RebaseAspNetCorePathBase = true) that paths work for the registration and login per tenant but if WithHostStrategy() it does not work.

When try to go to https://tenant.localhost/identity/account/login gives me a 404 error. I do have a tenant.localhost in the host file, so that is not the issue.

I dont know if this might be causing it but changed the database to use MsSql Server. I didnt get an error and did the migration correctly

I know this work but this is not for multi-tenant accounts but for normal identity setup.
```csharp
      services.ConfigureApplicationCookie(o => o.LoginPath = "/login");
      services.ConfigureApplicationCookie(o => o.LogoutPath = "/logout");
      services.ConfigureApplicationCookie(o => o.AccessDeniedPath = "/Account/AccessDenied");
```

I did read the ITenantInfo implements the specific paths but not working. Is there aways to make specific like in the code above?

Program.cs
```csharp
using Finbuckle.MultiTenant;
using Microsoft.AspNetCore.Identity;
using Microsoft.EntityFrameworkCore;
using IdentitySample.Data;

var builder = WebApplication.CreateBuilder(args);

// Add services to the container.
builder.Services.AddDbContext<ApplicationDbContext>();
builder.Services.AddDatabaseDeveloperPageExceptionFilter();

builder.Services.AddIdentity<ApplicationUser, IdentityRole>(opts =>
    {
        opts.SignIn.RequireConfirmedAccount = true;
    })
    .AddEntityFrameworkStores<ApplicationDbContext>();

builder.Services.AddRazorPages();

// Add MultiTenant
builder.Services.AddMultiTenant<TenantInfo>()
    .WithHostStrategy()
    .WithConfigurationStore()
    .WithPerTenantAuthentication();

var app = builder.Build();

// Apply migrations if needed
var store = app.Services.GetRequiredService<IMultiTenantStore<TenantInfo>>();
foreach (var tenant in await store.GetAllAsync())
{
    await using var db = new ApplicationDbContext(tenant);
    await db.Database.MigrateAsync();
}

// Configure the HTTP request pipeline.
if (app.Environment.IsDevelopment())
{
    app.UseMigrationsEndPoint();
}
else
{
    app.UseExceptionHandler("/Error");
    // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
    app.UseHsts();
}

app.UseHttpsRedirection();
app.UseStaticFiles();

app.UseMultiTenant();
app.UseRouting();

app.UseAuthentication();
app.UseAuthorization();

app.MapRazorPages();

app.Run();
``` 

I did try adding this code but resulted in a compile-time error
```csharp
        services.AddAuthentication(CookieAuthenticationDefaults.AuthenticationScheme)
            .AddCookie()
            .AddOpenIdConnect();
```

Would there be away to make so you could SSO? I don't want the users make accounts per tenant. 

I would like to have so when they create an account they create the TenantInfo configuration. So, basically this would be one-to-one relationship per user per tenant account.

The non-tenant (root) account has routes to Identity.

I will try to see if this routes correctly without using custom ApplicationUser entity.

**Comment by AndrewTriesToCode** [comment]
This should work. Just keeping the regular identity path options, does the login page load if you use just localhost with no subdomain?

**Comment by asparatu** [comment]
Hello Andrew,

It does not load even when just using localhost.

I am using your example project and just modifing it to work the why I wanted it. 

The privacy page works fine with or without the subdomains. I was thinking it might not be working because it might not registering the Microsoft.AspNet.Identity.UI library. I had something like happen because the route or/and page did not exist. 

I did try adding it with options extension and setting cookies loginpath but that didn't work also.

**Comment by asparatu** [comment]
Hello Andrew,

It does not load even when just using localhost. The privacy page works well, with or without the subdomains. I was initially unsure if it was functioning due to a possible issue with registering the Microsoft.AspNet.Identity.UI library. This has happened before when the route or page didn't exist. I am using your example project and just modifying it.

I tried to fix it using WithPerTenantOptions and setting CookieAuthenticationOptions, but this didn't fix it.
```csharp
builder.Services.AddMultiTenant<TenantInfo>()
    .WithHostStrategy()
    .WithConfigurationStore()
    .WithPerTenantAuthentication()
    .WithPerTenantOptions<CookieAuthenticationOptions>((opts, tenantInfo) =>
    {
        opts.Cookie.Name = $"{tenantInfo.Id}_{opts.Cookie.Name}";
        opts.LoginPath = "/Identity/Account/Login";
        opts.LogoutPath = "/Identity/Account/Logout";
        opts.Cookie.Path = $"/{tenantInfo.Identifier}";
    });
```

**Comment by asparatu** [comment]
Hello Andrew,

I found out why the routing was not working. There was problem with solution. I created a new MVC and Razor project and i didnt have any issue with it. 

I do need help on one more thing, I want to setup the Tenant Store to use EF. From the documentation you had sample project for asp.net 3.0 but it is not in the repo anymore.

could provide me sample project, please.

**Comment by AndrewTriesToCode** [comment]
Hi, I'm glad you got the routing to work for you. Regarding older samples, they are available in older version of the code such as here:

https://github.com/Finbuckle/Finbuckle.MultiTenant/tree/v6.9.1/samples

**Comment by asparatu** [comment]
Hello Andrew,

I discovered the reason behind the incorrect routing: it wasn't an issue with the project, but rather with the identity service itself.

When the identity service is configured in this manner, the default Identity routes from Microsoft.AspNetCore.Identity.UI will cease to function.
```csharp
builder.Services.AddIdentity<ApplicationUser, IdentityRole>(options => options.SignIn.RequireConfirmedAccount = true)
    .AddEntityFrameworkStores<ApplicationDbContext>();
```
If you have set up the Identity Service in this manner, there should be no problems with the routes.
```csharp
builder.Services.AddDefaultIdentity<ApplicationUser>(options => options.SignIn.RequireConfirmedAccount = true)
    .AddEntityFrameworkStores<ApplicationDbContext>();
```

I've discovered the reason behind this behavior. The utilization of **AddDefaultIdentity** introduces the UI components for the Razor pages. Conversely, opting for **AddIdentity** results in the removal of routing and UI related to Identity. This necessitates manual scaffolding of the UI components to ensure proper functionality, as outlined in this informative [article](https://stackoverflow.com/a/51289540) or you create your own razor pages to handle account management.
